crates/gpio/src/backend.rs

```diff
@@ -206,10 +237,15 @@ fn start_backend(args: GpioArgs) -> Result<()> {
                     let controller = PhysDevice::open(device_num).unwrap();
                     start_device_backend(controller, socket.clone());
                 }
+                #[cfg(feature = "dummy")]
                 GpioDeviceConfig::SimulatedDevice(num_gpios) => {
                     let controller = DummyDevice::open(num_gpios).unwrap();
                     start_device_backend(controller, socket.clone());
                 }
+                #[cfg(not(feature = "dummy"))]
```

Comment:
Again, special block like this can be avoided I guess. Maybe just fail DummyDevice::open() in this case, and let it panic.

Reply:
```
                #[cfg(not(feature = "mock_gpio"))]
                _ => {
                    return Err(Error::DeviceUnimplemented);
                }
```